core/arch/arm/pta/benchmark.c

```diff
@@ -43,6 +43,27 @@
 
 struct tee_ts_global *bench_ts_global;
 static struct mutex bench_reg_mu = MUTEX_INITIALIZER;
+static uint32_t prev_cntkctl;
```

Comment:
This variable and the three helper functions is a bit messy. I'd prefer something like:
```
static uint32_t timer_saved_ctrl;

static uint32_t timer_enable_el0_access(void)
{
    uint32_t ctrl = read_cntkctl();

    write_cntkctl(ctrl | CNTKCTL_PL0VCTEN);
    return ctrl;
}

static void timer_restore_access(uint32_t saved_ctrl)
{
    write_cntkctl(saved_ctrl);
}

Reply:
Such API will not work if there are non sequenced consumers for the generic timers access. Maybe a generic support would require some reference counting.

Reply:
True, but as far as I understand this is only used in sequence. By the way, that's why I want the caller of `timer_enable_el0_access()` etc to the be one handling the value to be restored later since it can't be easily abstracted.

Reply:
Fixed, thanks